### PR TITLE
add downloader for Der Standard

### PR DESCRIPTION
**Pull request by iNtEgraIR2021**
Hi @thisisparker 

Following the workflow used in #76 I also created a downloader for the crosswords published by ['Der Standard'](https://www.derstandard.at/lifestyle/raetsel-sudoku/kreuzwortraetsel) a major news outlet from Austria.

Looking forward to your feedback.

Regards,
Petra

### CODE REVIEW

**Review by thisisparker — CHANGES_REQUESTED** [review]
Left some comments on here, but at a first glance this looks good and I appreciate you working out some new approaches.

File: README.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ Supported outlets:
 |*Vox*|`vox`|✔️|||
 |*Wall Street Journal*|`wsj`|✔️||✔️|
 |*Washington Post*|`wp`|✔️|✔️||
+|*Der Standard*|`std`|✔️||✔️|
```
[the remark this answers is not in the document]
Comment (iNtEgraIR2021) [reply README.md:29]: Ok done

File: xword_dl/downloader/derstandarddownloader.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            'Ö':'UE',
+            'ä':'ae',
+            'ö':'oe',
+            'ü':'üe',
+            'ß':'ss'
+        }
+
+        title = str(puzzle.title)
+        for umlaut, rep in umlauts.items():
+            title = re.sub(re.escape(umlaut), re.escape(rep), title)
```
[the remark this answers is not in the document]
Comment (iNtEgraIR2021) [reply xword_dl/downloader/derstandarddownloader.py:92]: Ok I removed `re` and used your suggestion for the replacement.

File: xword_dl/downloader/derstandarddownloader.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import datetime
+import json
+import urllib
```
Comment (thisisparker) [inline xword_dl/downloader/derstandarddownloader.py:3]: Needs an `import re` between these two lines. Looks like that got clipped during review, sorry!

File: xword_dl/downloader/derstandarddownloader.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        index_res = requests.get(index_url, headers=self.headers, timeout=10)
+        index_soup = BeautifulSoup(index_res.text, "html.parser")
+
+        latest_fragment = next(a for a in index_soup.select('.teaser-inner > a'))['href']
+        latest_absolute = urllib.parse.urljoin('https://www.derstandard.at',
+                                               latest_fragment)
+
+        landing_page_url = latest_absolute
+
+        print(landing_page_url)
```
Comment (thisisparker) [inline xword_dl/downloader/derstandarddownloader.py:42]: Please remove this print (although useful for debugging... maybe I should support some kind of verbosity switch! but not today)

File: xword_dl/downloader/derstandarddownloader.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        res = requests.get(url, headers=self.headers, timeout=10)
+
+        try:
+            res.raise_for_status()
+        except requests.exceptions.HTTPError:
+            raise XWordDLException('Unable to load {}'.format(url))
+
+        try:
+            # html embed content is encoded -> beautifulsoup parsing would not work
+            query_id = list(re.findall(r'(http)(s)*(:\/\/cdn\-eu1\.amuselabs\.com\/pmm\/crossword)(\?id\=)(bd74ab46)', str(res.text)))
```
Comment (thisisparker) [inline xword_dl/downloader/derstandarddownloader.py:56]: This is overmatched to a single ID. To match in general it should have (I think) `([0-9a-f]{8})` as the last group.

File: xword_dl/downloader/derstandarddownloader.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+            if len(query_id) == 0:
+                raise XWordDLException('Cannot find puzzle at {} -> failed to retrieve amuselabs url from encoded html.'.format(url))
+            
+            self.id = str(query_id[0][-1])
+        except KeyError:
+            raise XWordDLException('Cannot find puzzle at {}.'.format(url))
+
+        return self.find_puzzle_url_from_id(self.id)
+        
+
```
Comment (thisisparker) [inline xword_dl/downloader/derstandarddownloader.py:67]: Just for consistency's sake, nitpicky spacing preferences of 1 blank line between methods

**Review by thisisparker — CHANGES_REQUESTED** [review]
Looking good, very close, just a few more tiny things with the latest changes

File: xword_dl/downloader/derstandarddownloader.py (hunk 5 above)
Comment (iNtEgraIR2021) [reply xword_dl/downloader/derstandarddownloader.py:56]: Good catch

**Comment by iNtEgraIR2021** [comment]
Tried to apply everything. Hope I didn't miss anything. Looking forward to your feedback

**Review by thisisparker — APPROVED** [review]

**Comment by thisisparker** [comment]
Thank you! Got it merged, and it'll ship with the next release! :tada:

**Pull request merged**